cog/task.py

```diff
@@ -148,21 +148,25 @@ def git_clone(url, sha, target=None, work_dir=None, log=False):
 
     target = os.path.abspath(target)
 
+    # If the target does not exist, clone it.
     if not os.path.exists(target):
         cmd = ' '.join(['git clone', url, target, '&& cd %s && ' % target,
                        'git checkout', sha, '&> clone.log'])
-        rc = system(cmd)
-
-        if log:
-            with open(os.path.join(target, 'clone.log')) as f:
-                clone_log = f.read()
-            return rc, clone_log
 
-        return rc
+    # If the target does exist, change into it and attempt to checkout the sha.
```

Comment:
Does the target ever exist? I assume tasks would run in a new directory each time - this could have issues with reproducibility if there are lingering files in the repo not tracked by git.

Reply:
That is a good point. For the automatic tools when each PR is submitted, then yes it makes sense to have a new directory each time. However, when testing tasks, it was convenient to not have the repository get re-cloned each time since that takes a while. All I had to do was modify the task itself and rerun it.

That being said, since it only affects testing really, if you think it's best I can remove this change for now. I agree that if the directory already exists, then maybe it could cause problems. The first change with `args` is more important.

Reply:
In that case its fine to leave in, but maybe add a warning message noting that it's reusing an existing git repo.